packages/klighd-core/src/search-bar/searchbar-actions.ts

```diff
@@ -346,8 +352,8 @@ export class SearchBarActionHandler implements IActionHandler {
      * Remove all highlights
      * @param results the highlighted results
      */
-    private removeHighlights(panel: SearchBarPanel): void {
-        for (const result of panel.getResults) {
+    private removeHighlights(searchResults: SearchResult[]): void {
+        for (const result of searchResults) {
```

Comment:
I think this is more unintuitive. Removing all highlights should not require the action sender to know what the highlights are, because this UI extension should remember that. Why did you change it here?

Reply:
The Action Sender in this case is the panel. The panel stores the search results and the contract is that all search results are always highlighted. 

The previous implementation would in some situations send the clear highlights action and then immediately empty the search results, which would result in elements remaining highlighted. By passing responsibility for clearing the list to action handler we avoid this. Of course we could remove the parameter from the handler again and access the list through this.panel but in my mind this seemed clearer.